hw3/02-charts.js

```diff
@@ -26,27 +28,93 @@ const borderColors = [
   'rgba(78, 52, 199, 1)',
 ];
 
-// url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+// Function to fetch Thrones API data
+const fetchCharacters = async () => {
+  try{
+  const res = await fetch('https://thronesapi.com/api/v2/Characters');
+  const data = await res.json();
+  return data;
+  } 
+  catch (error) {
+  console.error("There was an error fetching the characters:", error);
+  return [];
+  }
+};
+
+// Function to sanitize house names
+const sanitizeHouseName = (name) => {
+  switch (name) {
+    case 'Mormont':
+      return 'House Mormont';
+    case 'Baratheon':
+      return 'House Baratheon';
+    case 'Bolton':
+      return 'House Bolton';
+    case 'Stark':
+      return 'House Stark';
+    case 'Targaryan':
+    case 'Targaryen':
+      return 'House Targaryen';
+    case 'House Lannister':
+    case 'Lannister':
+    case 'Lanister':
+    case 'House Lanister':
+      return 'House Lannister';
+    case 'Tarth':
+      return 'House Tarth';
+    case 'Greyjoy':
+      return 'House Greyjoy';
+    case '':
+    case 'None':
+    case 'Unkown':
+    case undefined:
+      return 'Unknown';
+    default:
+      return name;
+  }
+};
+
+const countByHouse = (characters) => {
+  const counts = {};
+
+  characters.forEach((character) => {
+    const house = sanitizeHouseName(character.family);
+    if (house) {
+      counts[house] = (counts[house] || 0) + 1;
+    }
+  });
+  return counts;
+};
 
-const renderChart = () => {
+const renderChart = (houseData) => {
+  const labels = Object.keys(houseData);
+  const data = Object.values(houseData);
   const donutChart = document.querySelector('.donut-chart');
 
   new Chart(donutChart, {
     type: 'doughnut',
     data: {
-      labels: ['label', 'label', 'label', 'label'],
+      labels,
       datasets: [
         {
-          label: 'My First Dataset',
-          data: [1, 12, 33, 5],
+          data,
           backgroundColor: backgroundColors,
           borderColor: borderColors,
           borderWidth: 1,
         },
       ],
     },
+    options: {
+      legend: {
+        display: false
```

Comment:
I understand not wanting a legend in some cases, however with so many different groups, it becomes hard to read interpret the data at a glance. I saw you mention that you do have not seen game of thrones, but I think some grouping of the 1 person groups could be helpful.